libcst/codemod/commands/apply_stub_annotations.py

```diff
@@ -0,0 +1,506 @@
+# Copyright (c) 2016-present, Facebook, Inc.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree
+
+
+from typing import (
+    Dict,
+    List,
+    NamedTuple,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    Union,
+    overload,
+)
+
+import libcst as cst
+
+
+def _get_attribute_as_string(attribute: cst.BaseExpression) -> str:
+    names = []
+    while isinstance(attribute, cst.Attribute):
+        # pyre-fixme[16]: `BaseExpression` has no attribute `value`.
+        if isinstance(attribute.value.value, cst.Attribute):
+            value = _get_attribute_as_string(
+                cst.ensure_type(attribute.value, cst.Attribute).value
+            )
+        else:
+            value = _get_name_as_string(attribute.value.value)
+        names.append(value)
+        attribute = attribute.attr
+    if attribute is not None:
+        names.append(_get_name_as_string(attribute.value))
+    return ".".join(names)
+
+
+def _get_name_as_string(node: Union[cst.CSTNode, str]) -> str:
+    if isinstance(node, cst.Name):
+        return node.value
+    else:
+        # pyre-fixme[7]: Expected `str` but got `Union[CSTNode, str]`.
```

Comment:
Are the pyre-fixme's in the original? I agree with the typechecker on many of them, like this one.

Reply:
Yes, they were in the original. Sorry about this. I'm unfamiliar with this code too, but I'll try to fix as many as I can.